NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# Cb edit current user posts v2 ticket8

Description:
Trello ticket 8: https://trello.com/c/HvNR09km/8-8-edit-a-post
This ticket allows the logged in user to to edit their own posts :

Given an author is viewing a specific Post that they have written OR the list of Posts
When they select the edit option for a Post they have authored
Then the user should be directed to a form and given the ability to change the Post’s content

Given the author wishes to confirm the edits
When they the select the option to confirm
Then the Post saved to the database via UPDATE
And the author should be directed back to the Post Details for this Post

Given the author decides not to confirm the Edit
When they the select the option to reject confirmation
Then the Post should NOT show any Edits
And the author should be directed back to the Post list


Testing Instructions:

1. git add and git commit whatever you're working on on your branch.
2. git checkout main.
3. git fetch -a.
4. git checkout my branch: cb-editCurrentUserPostsV2-ticket8.
5. If you don't see the most recent version of my files, git pull origin cb-editCurrentUserPostsV2-ticket8.
6. (if not already done) Please open the TabloidFullStack.sln with Visual Studio 2022.
7. launch the API by clicking the execute with debugger solid green button in Visual Studio 2022.
8. (if not already open)Open a new GitBash window and CD into the tabloid file inside the client folder of the project and open Visual Studio Code with the command "code ". Then launch the react app with the command "npm start".
9. In the react app in the browser, if you are not already logged in, you should be directed to the login page. Please log in with this email address: foo@bar.com. This should redirect you back to the home page.
10. On nav bar, click "My Posts". You should be directed to the path "http://localhost:3000/my-posts", and see all the posts for that user. 
11. Click on the title on the top post to open the post details. Make edit to the post, but instead of clicking "Save", click "Cancel". Please verify that you are directed back to the list of posts.
12. Click again on the title on the top post to open the post details. Make edit to the post and this time, click "Save". Please verify that you are directed to the the post details again for that post and that your changes are reflected on the post. (Be aware that while "My Posts" will show a list of all posts by the current user, "Posts" will only display posts that include a publish datetime.)

Checklist:
